### PR TITLE
migrate_options_shared: Add hpt resize PPC case

### PR DESCRIPTION
**Pull request by dzhengfy**
Signed-off-by: Dan Zheng <dzheng@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/migration/migrate_options_shared.py [hunk 1]
```diff
@@ -304,11 +389,25 @@ def check_timeout_postcopy(params):
             libvirtd_conf = libvirt.customize_libvirt_config(libvirtd_conf_dict,
                                                              remote_host=True,
                                                              extra_params=params)
+        if hpt_resize:
+            new_xml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
+            set_hpt(hpt_resize, new_xml)
```
[the remark this answers is not in the document]
Comment (dzhengfy) [reply libvirt/tests/src/migration/migrate_options_shared.py]: Yeah, this should be refined. Thanks. Done.

**Comment by dzhengfy** [comment]
Test result on Power 8: 
(1/6) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.hpt_resize.enabled.with_postcopy: PASS (245.72 s)
 (2/6) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.hpt_resize.enabled.without_postcopy: PASS (244.29 s)
 (3/6) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.hpt_resize.disabled.with_postcopy: PASS (245.00 s)
 (4/6) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.hpt_resize.disabled.without_postcopy: PASS (244.57 s)
 (5/6) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.hpt_resize.required.with_postcopy: PASS (245.03 s)
 (6/6) type_specific.io-github-autotest-libvirt.virsh.migrate_options_shared.positive_test.hpt_resize.required.with_postcopy: PASS (237.17 s)

File: libvirt/tests/src/migration/migrate_options_shared.py [hunk 2]
```diff
@@ -62,16 +63,93 @@ def setup_libvirtd_conf_dict(params):
...
+    """
+
+    def set_hpt(resizing, vmxml):
+        """
+        Set the hpt feature for PPC
+
+        :param resizing: The value needed
+        :param vmxml: guest xml
+
+        """
```
Comment (balamuruhans) [inline libvirt/tests/src/migration/migrate_options_shared.py]: last blank lines can be removed here and else where

File: libvirt/tests/src/migration/migrate_options_shared.py [hunk 3]
```diff
@@ -62,16 +63,93 @@ def setup_libvirtd_conf_dict(params):
...
+        hpt_order_path = "/sys/kernel/debug/powerpc/hpt_order"
+        hpt_order = session.cmd_output('cat %s' % hpt_order_path).strip()
+        hpt_order = int(hpt_order)
+        logging.info('Current hpt_order is %d', hpt_order)
+        hpt_order += 1
+        cmd = 'echo %d > %s' % (hpt_order, hpt_order_path)
+        cmd_result = session.cmd_status_output(cmd)
+        result = process.CmdResult(stderr=cmd_result[1],
```
Comment (balamuruhans) [inline libvirt/tests/src/migration/migrate_options_shared.py]: why do we use this ?

File: libvirt/tests/src/migration/migrate_options_shared.py [hunk 4]
```diff
@@ -62,16 +63,93 @@ def setup_libvirtd_conf_dict(params):
...
+
         """
-        The operations to the VM need to be done before migration happens
+        cmd = 'ps -ef|grep qemu|grep -v grep'
+        qemu_line = results_stdout_52lts(process.run(cmd, shell=True))
+        if content not in qemu_line:
+            if err_ignore:
+                return False
+            else:
+                test.fail("Expected '%s' was not found in qemu command line",
```
Comment (balamuruhans) [inline libvirt/tests/src/migration/migrate_options_shared.py]: `test.fail("Expected '%s' was not found in qemu command line" % content)`

File: libvirt/tests/src/migration/migrate_options_shared.py (hunk 3 above)
Comment (dzhengfy) [reply libvirt/tests/src/migration/migrate_options_shared.py]: I think this is to leverage 'libvirt.check_exit_status()' to verify the result.

File: libvirt/tests/src/migration/migrate_options_shared.py (hunk 4 above)
Comment (dzhengfy) [reply libvirt/tests/src/migration/migrate_options_shared.py]: Done

File: libvirt/tests/src/migration/migrate_options_shared.py (hunk 3 above)
Comment (balamuruhans) [reply libvirt/tests/src/migration/migrate_options_shared.py]: got it. I think this the better way.

**Comment by balamuruhans** [comment]
@dzhengfy  Please rebase it.

File: libvirt/tests/src/migration/migrate_options_shared.py [hunk 5]
```diff
@@ -63,16 +64,89 @@ def setup_libvirtd_conf_dict(params):
...
-        logging.info("Check local VM network connectivity before migrating")
-        s_ping, o_ping = utils_test.ping(vm.get_address(), count=10, timeout=20)
-        logging.info(o_ping)
+        # Confirm local/remote VM can be accessed through network.
+        logging.info("Check VM network connectivity")
+        s_ping, o_ping = utils_test.ping(vm.get_address(),
+                                         count=10,
+                                         timeout=20,
+                                         output_func=logging.debug,
+                                         session=session)
```
[the remark this answers is not in the document]
Comment (dzhengfy) [reply libvirt/tests/src/migration/migrate_options_shared.py]: From my run log, I do already see the ping logs which is provided by utils_test.ping(...output_func=logging.debug...), so we need not the duplicated logs, I think.

File: libvirt/tests/src/migration/migrate_options_shared.py [hunk 6]
```diff
@@ -63,16 +64,89 @@ def setup_libvirtd_conf_dict(params):
...
-        The operations to the VM need to be done before migration happens
+        cmd = 'ps -ef|grep qemu|grep -v grep'
+        qemu_line = results_stdout_52lts(process.run(cmd, shell=True))
+        if content not in qemu_line:
+            if err_ignore:
+                return False
+            else:
+                test.fail("Expected '%s' was not found in "
+                          "qemu command line" % content)
+        return True
```
[the remark this answers is not in the document]
Comment (dzhengfy) [reply libvirt/tests/src/migration/migrate_options_shared.py]: https://github.com/avocado-framework/avocado-vt/pull/1652

[the remark this answers is not in the document]
Comment (dzhengfy) [reply libvirt/tests/src/migration/migrate_options_shared.py]: Depends on https://github.com/avocado-framework/avocado-vt/pull/1652

File: libvirt/tests/src/migration/migrate_options_shared.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (kylazhang) [reply libvirt/tests/src/migration/migrate_options_shared.py]: I mean if not, use _ instead

**Comment by iccaszhulili** [comment]
rh-caselink retest please

File: libvirt/tests/src/migration/migrate_options_shared.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (dzhengfy) [reply libvirt/tests/src/migration/migrate_options_shared.py]: Fixed. Thanks.

**Pull request merged**